lib/cache/expirations/AbsoluteExpiration.ts

```diff
@@ -0,0 +1,27 @@
+import { Expiration } from './Expiration';
+
+export class AbsoluteExpiration implements Expiration {
+
+  private readonly expirations = new Set<string>();
+
+  constructor(
+    private readonly timeout: number,
+  ) { }
+
+  public add(key: string, clear: (key: string) => unknown): void {
+    if (this.expirations.has(key)) {
+      return;
+    }
+
+    this.expirations.add(key);
+    setTimeout(this.clear(key, clear), this.timeout);
```

Comment:
Simplify code, don't return a function in `clear` and use directly: `setTimeout(() => this.clear(key, clear), this.timeout);`